Code/C++/Doubly_linkedlist/Doubly_linkedlist_all_operations.cpp

```diff
@@ -0,0 +1,390 @@
+    cout << "\nEnter the location where you want to insert new node";
+    cin >> index;
+    if (index < count)
+    {
+        cout << "\nInvalid position ";
+    }
+    else if (start == nullptr || last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (index == 4)
+    {
+        insertion_at_end();
+    }
+    else
+    {
+        while (count != index)
+        {
+            ptr = ptr->next;
+            count += 1;
+        }
+        new_node->next = ptr->next;
+        ptr->next->prev = new_node;
+        ptr->next = new_node;
+        new_node->prev = ptr;
+    }
+}
+
+//Function to insert a node at given index.
+
+void Doubly_linklist::insertion_at_index()
+{
+    node *new_node = new node;
+    node *ptr = start;
+    cout << "\nEnter the info of the node you want to insert";
+    cin >> new_node->data;
+    int count = 0, index;
+    cout << "\nEnter the location where you want to insert new node";
+    cin >> index;
+    if (index < count)
+    {
+        cout << "\nInvalid position ";
+    }
+    else if (start == nullptr || last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (index == 0)
+    {
+        insertion_at_beginning();
+    }
+    else
+    {
+        while (count != index - 1)
+        {
+            ptr = ptr->next;
+            count += 1;
+        }
+        new_node->next = ptr->next;
+        ptr->next->prev = new_node;
+        ptr->next = new_node;
+        new_node->prev = ptr;
+    }
+}
+
+//Fuction for deleting a node from beginning.
+
+void Doubly_linklist::deletion_at_beginning()
+{
+    node *ptr = start;
+    if (start == nullptr && last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (start->next == nullptr)
+    {
+        start = nullptr;
+        last = nullptr;
+        free(start);
+        free(last);
+    }
+    else
+    {
+        start = start->next;
+        start->prev = nullptr;
+        ptr->next = nullptr;
+    }
+}
+
+//Fuction for deleting a node from end.
+
+void Doubly_linklist::deletion_at_end()
+{
+    node *temp = last;
+    if (start == nullptr && last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
+    else if (start->next == nullptr)
+    {
+        start = nullptr;
+        last = nullptr;
+        free(start);
+        free(last);
+    }
+    else
+    {
+        last = last->prev;
+        last->next = nullptr;
+        temp->prev = nullptr;
+        free(temp);
+    }
+}
+
+//Fuction for deleting a node from any index.
+
+void Doubly_linklist::deletion_at_index()
+{
+    node *ptr = start;
+    node *ptr_1 = start->next;
+    int index;
+    cout << "\nEnter the index where you want to delete the node ";
+    cin >> index;
+    int count = 0;
+    if (index == 0)
+    {
+        deletion_at_beginning();
+    }
+    else if (index < 0)
+    {
+        cout << "\nInvalid index ";
+    }
+    else if (index == 4)
+    {
+
+        deletion_at_end();
+    }
+    else
+    {
```

Comment:
Here also, add a similar check as before, i.e. index should not exceed the size of the list - 1.